Handle JIRA Error during history extraction

bockthom opened the pull request:
When extracting changelog information for a JIRA issue via the corresponding API, it can happen that the changelog information for this issue is not available (for whatever reason) und cannot be extracted. In such a case, the API call ends up in a JIRA Error. If this happens, just catch this error, print a warning, and create an empty history, so that no further events are processed in this case.

bockthom committed:
Handle JIRA Error during history extraction

When extracting changelog information for a JIRA issue via the
corresponding API, it can happen that the changelog information for this
issue is not available (for whatever reason) und cannot be extracted. In
such a case, the API call ends up in a JIRA Error. If this happens, just
catch this error, print a warning, and create an empty history, so that
no further events are processed in this case.

Signed-off-by: Thomas Bock <bockthom@cs.uni-saarland.de>

hechtlC commented on issue_processing/jira_issue_processing.py:
```diff
@@ -403,108 +404,116 @@ def load_issue_via_api(issues, persons, url):
+                # default values for state and resolution
+                old_state, new_state, old_resolution, new_resolution = "open", "open", "unresolved", "unresolved"
+
+                # all changes in the issue changelog are checked if they contain an useful information
```
Since you are already fixing a/an errors above: It is a useful... not an useful...

bockthom replied:
As comments and documentation were inconsistently formatted in general, I decided to streamline all the comments in this file. I hope you are fine with that :smile: 

(See c08c7a1)

hechtlC reviewed (approved):
Looks good. Thanks for the fix @bockthom! I just found one tiny spelling mistake but once that is fixed you can merge the PR.

bockthom committed:
Streamline comments and documentation in JIRA issue processing

- Let all sentences in a function documentation consistently end with a
period.
- Let all sentences in a function documentation consistently use
the imperative instead of third person view.
- Fix grammar in comments and documentation.

Signed-off-by: Thomas Bock <bockthom@cs.uni-saarland.de>
